src/main/java/org/kiwiproject/jersey/client/ServiceIdentifier.java

```diff
@@ -0,0 +1,129 @@
+package org.kiwiproject.jersey.client;
+
+import static java.util.Objects.isNull;
+import static org.kiwiproject.base.KiwiPreconditions.checkArgumentNotNull;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+import java.time.Duration;
+
+@SuppressWarnings("WeakerAccess")
+@NoArgsConstructor
+@Getter
+@Setter
+@ToString
+public class ServiceIdentifier {
+    public enum Connector {
+        ADMIN, APPLICATION
+    }
+
+    private String serviceName;
+    private String preferredVersion;
+    private String minimumVersion;
+
+    private Connector connector = Connector.APPLICATION;
+    private Duration connectTimeout = RegistryAwareClientConstants.DEFAULT_CONNECT_TIMEOUT;
+    private Duration readTimeout = RegistryAwareClientConstants.DEFAULT_CONNECT_REQUEST_TIMEOUT;
+
+    public ServiceIdentifier(String serviceName) {
+        this(serviceName, null, Connector.APPLICATION);
+    }
+
+    public ServiceIdentifier(String serviceName, String preferredVersion) {
+        this(serviceName, preferredVersion, Connector.APPLICATION);
+    }
+
+    public ServiceIdentifier(String serviceName, Connector connector) {
+        this(serviceName, null, connector);
+    }
+
+    public ServiceIdentifier(String serviceName, String preferredVersion, Connector connector) {
+        this.serviceName = serviceName;
+        this.preferredVersion = preferredVersion;
+        this.connector = connector;
+    }
+
+    public ServiceIdentifier(String serviceName, Duration connectTimeout, Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName, Connector connector, Duration connectTimeout, Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.connector = connector;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName,
+                             String preferredVersion,
+                             Connector connector,
+                             Duration connectTimeout,
+                             Duration readTimeout) {
+        this.serviceName = serviceName;
+        this.preferredVersion = preferredVersion;
+        this.connector = connector;
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+    }
+
+    public ServiceIdentifier(String serviceName,
```

Comment:
Perhaps now would be a good time to just use a (Lombok) builder instead of all these telescoping constructors?